did/plugins/zammad.py

```diff
@@ -61,7 +63,25 @@ def search(self, query):
         try:
             result = result["Ticket"]
         except KeyError:
-            result = dict()
+            result = {}
+        log.debug("Result: {0} fetched".format(listed(len(result), "item")))
+        log.data(pretty(result))
+        return result
+
+    def get_articles(self, ticket_id):
+        """ Perform Zammad query """
+        url = self.url + "/ticket_articles/by_ticket/" + str(ticket_id)
+        log.debug("Zammad query: {0}".format(url))
+        try:
+            request = urllib.request.Request(url, headers=self.headers)
+            response = urllib.request.urlopen(request)
+            log.debug("Response headers:\n{0}".format(
+                str(response.info()).strip()))
+        except urllib.error.URLError as error:
+            log.debug(error)
+            raise ReportError(
+                "Zammad search on {0} failed.".format(self.url))
```

Comment:
This part seems to be copy & pasted from the `search()` method. What about separating the common code to prevent duplication? There could be a common `search()` method used by both `get_articles()` and `get_tickets()`. What do you think, @come-nc?